### ISSUE TITLE
rmsprop or rprop?

### ISSUE THREAD

**Issue by m3rik** [Microsoft/CNTK#1298]
Hello,

I think that in cntk the rprop algorithm is implemented. Rmsprop is a better variant that does not use so many hyperparameters.
Hope I'm not wrong.

Rprop paper:
http://www.inf.fu-berlin.de/lehre/WS06/Musterererkennung/Paper/rprop.pdf

Rmsprop was introduced by Hinton on coursera during a lecture, but it is very well explained here:
https://i.imgur.com/q1noU76.png

Have a good day,
Paul

**Comment by cha-zhang** [comment]
What's implemented in CNTK is actually a hybrid of Rprop and RMSProp. We do normalize the gradient with the decaying average over past squared gradient. 

It's arguable whether the learning rate adaptation is necessary, though it should not hurt performance. We might some day change the code to be the standard RMSProp, if majority of the users want that.

**Comment by m3rik** [comment]
Thank you very much!
